src/python/pants/backend/python/rules/hermetic_pex.py

```diff
@@ -1,34 +1,161 @@
 # Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
-
-from typing import Any, Iterable, Mapping, Optional
+import logging
+import os
+from dataclasses import dataclass
+from typing import Any, Iterable, Mapping, Optional, Tuple
 
 from pants.backend.python.subsystems.subprocess_environment import SubprocessEncodingEnvironment
+from pants.engine import process
+from pants.engine.engine_aware import EngineAware
 from pants.engine.fs import Digest
-from pants.engine.process import Process
+from pants.engine.process import BinaryPathRequest, BinaryPaths, Process
+from pants.engine.rules import collect_rules, rule
+from pants.engine.selectors import Get, MultiGet
 from pants.python.python_setup import PythonSetup
+from pants.subsystem.subsystem import Subsystem
+from pants.util.logging import LogLevel
+from pants.util.memo import memoized_property
+from pants.util.ordered_set import OrderedSet
 from pants.util.strutil import create_path_env_var
 
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class PexEnvironment(EngineAware):
+    path: Iterable[str]
+    interpreter_search_paths: Iterable[str]
+    bootstrap_python: Optional[str] = None
+
+    def create_argv(self, pex_path: str, *args: str) -> Iterable[str]:
+        argv = [self.bootstrap_python] if self.bootstrap_python else []
+        argv.extend((pex_path, *args))
+        return argv
+
+    @property
+    def environment_dict(self) -> Mapping[str, str]:
+        return dict(
+            PATH=create_path_env_var(self.path),
+            PEX_PYTHON_PATH=create_path_env_var(self.interpreter_search_paths),
+        )
+
+    def level(self) -> Optional[LogLevel]:
+        return LogLevel.INFO if self.bootstrap_python else LogLevel.WARN
+
+    def message(self) -> Optional[str]:
+        if not self.bootstrap_python:
+            return (
+                "No bootstrap python executable could be found. "
+                "Will attempt to run PEXes directly."
+            )
+        else:
+            return f"Selected {self.bootstrap_python} to bootstrap PEXes with."
+
+
+class PexRuntimeEnvironment(Subsystem):
+    options_scope = "pex"
+
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--binary-search-path",
```

Comment:
Putting this on a separate `Process` specific subsystem would help to head in the direction specified in #9760 (ie that the PATH is something that can be globally configured and locally filtered), but also be possible to do via a deprecation, so not a blocker.

Reply:
Going to punt on this.